README.md

```diff
@@ -1512,6 +1512,52 @@ would enable the use of a `@fold` on the `adjacent_animal` vertex field of `Foo`
 }
 ```
 
+## Schema Generation
+To generate a GraphQL schema, we first map a database schema into an intermediate representation:
+the `SchemaGraph`.
+
+#### SchemaGraph
+* `SchemaGraph`
+    * Represents a database's schema through `SchemaElement` objects.
+        * `SchemaElement`
+            * Can be one of three kinds:
+                * `EdgeType`
+                * `VertexType`
+                * `NonGraphElement`.
+            * Have the following attributes:
+                * `name`
+                * `abstract`
+                * `properties`, which are represented using GraphQLTypes.
+                * `in_connections`/`out_connections`, which represent the set of possible adjacent 
+                  `SchemaElements`.
+                    * For an `EdgeType`, its connections represent the set of allowed `EdgeTypes` 
+                      and `NonGraphElements` allowed at each end of the edge.
+                    * For `VertexType` and `NonGraphElement` objects, their connections represent
+                      the set of edges connected to the `SchemaElement`. A `NonGraphElement` may 
+                      only have connections if it's abstract and all its non-abstract subclasses 
+                      are `VertexTypes`. 
+            * `EdgeType` objects also have:
+                * `base_in_connection`/`base_out_connection`, which are the class allowed at each 
+                  edge end that is the superclass of all other classes allowed at the edge end.
+    * Keeps track of a schema's transitive inheritance structure.
+    * Keeps track of a schema's indexes. Indexes are not used in the GraphQL schema generation and 
```

Comment:
Nit: put these at the top, above SchemaElement. It's impossible to see what level of indentation these bullets are at.

Reply:
Will fix.